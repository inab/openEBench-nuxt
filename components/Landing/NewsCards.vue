<template>
	<v-container>
		<h2 class="text-h4 mb-10">What's new?</h2>

		<v-row>
			<v-col
				v-for="(item, index) in blogData"
				:key="index"
				align="center"
				sm="12"
				md="3"
			>
				<news-card
					:title="item.title"
					:desc="item.desc"
					img="https://source.unsplash.com/random"
				/>
			</v-col>
		</v-row>
	</v-container>
</template>

<script>
import NewsCard from '~/components/Landing/NewsCard';

export default {
	name: 'NewsCards',
	components: { NewsCard },
	data() {
		return {
			blogData: [
				{
					title: 'Heading',
					desc: 'This is a media card. You can use this section to describe the content.',
				},
				{
					title: 'Heading',
					desc: 'This is a media card. You can use this section to describe the content.',
				},
				{
					title: 'Heading',
					desc: 'This is a media card. You can use this section to describe the content.',
				},
				{
					title: 'Heading',
					desc: 'This is a media card. You can use this section to describe the content.',
				},
			],
		};
	},
};
</script>
