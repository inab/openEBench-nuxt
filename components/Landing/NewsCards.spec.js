import { mount } from '@vue/test-utils';
import NewsCards from './NewsCards.vue';

const factory = () => {
	return mount(NewsCards, {
		...createComponentMocks({}),
		mocks: {
			$vuetify: { breakpoint: {} },
		},
	});
};

describe('NewsCards', () => {
	it('is instantiated', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
	});

	it('should match snapshot', () => {
		const wrapper = factory();
		expect(wrapper.html()).toMatchSnapshot();
	});
});
