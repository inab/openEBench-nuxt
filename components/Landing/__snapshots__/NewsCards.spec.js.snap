// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCards should match snapshot 1`] = `
"<div class=\\"container\\">
  <h2 class=\\"text-h4 mb-10\\">What's new?</h2>
  <div class=\\"row\\">
    <div align=\\"center\\" class=\\"col-sm-12 col-md-3 col\\">
      <div class=\\"mx-auto v-card v-sheet theme--light\\" style=\\"max-width: 400px;\\">
        <div class=\\"v-image v-responsive white--text theme--light\\" style=\\"height: 200px;\\">
          <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
            <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
          </transition-stub>
          <div class=\\"v-responsive__content\\"></div>
        </div>
        <div class=\\"v-card__title\\">
          Heading
        </div>
        <div class=\\"v-card__text text-left\\"><span class=\\"text--primary\\">
			This is a media card. You can use this section to describe the content.
		</span></div>
        <div class=\\"v-card__actions\\"><button type=\\"button\\" class=\\"v-btn v-btn--text theme--light v-size--default primary--text\\"><span class=\\"v-btn__content\\"> Read more </span></button></div>
      </div>
    </div>
    <div align=\\"center\\" class=\\"col-sm-12 col-md-3 col\\">
      <div class=\\"mx-auto v-card v-sheet theme--light\\" style=\\"max-width: 400px;\\">
        <div class=\\"v-image v-responsive white--text theme--light\\" style=\\"height: 200px;\\">
          <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
            <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
          </transition-stub>
          <div class=\\"v-responsive__content\\"></div>
        </div>
        <div class=\\"v-card__title\\">
          Heading
        </div>
        <div class=\\"v-card__text text-left\\"><span class=\\"text--primary\\">
			This is a media card. You can use this section to describe the content.
		</span></div>
        <div class=\\"v-card__actions\\"><button type=\\"button\\" class=\\"v-btn v-btn--text theme--light v-size--default primary--text\\"><span class=\\"v-btn__content\\"> Read more </span></button></div>
      </div>
    </div>
    <div align=\\"center\\" class=\\"col-sm-12 col-md-3 col\\">
      <div class=\\"mx-auto v-card v-sheet theme--light\\" style=\\"max-width: 400px;\\">
        <div class=\\"v-image v-responsive white--text theme--light\\" style=\\"height: 200px;\\">
          <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
            <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
          </transition-stub>
          <div class=\\"v-responsive__content\\"></div>
        </div>
        <div class=\\"v-card__title\\">
          Heading
        </div>
        <div class=\\"v-card__text text-left\\"><span class=\\"text--primary\\">
			This is a media card. You can use this section to describe the content.
		</span></div>
        <div class=\\"v-card__actions\\"><button type=\\"button\\" class=\\"v-btn v-btn--text theme--light v-size--default primary--text\\"><span class=\\"v-btn__content\\"> Read more </span></button></div>
      </div>
    </div>
    <div align=\\"center\\" class=\\"col-sm-12 col-md-3 col\\">
      <div class=\\"mx-auto v-card v-sheet theme--light\\" style=\\"max-width: 400px;\\">
        <div class=\\"v-image v-responsive white--text theme--light\\" style=\\"height: 200px;\\">
          <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
            <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
          </transition-stub>
          <div class=\\"v-responsive__content\\"></div>
        </div>
        <div class=\\"v-card__title\\">
          Heading
        </div>
        <div class=\\"v-card__text text-left\\"><span class=\\"text--primary\\">
			This is a media card. You can use this section to describe the content.
		</span></div>
        <div class=\\"v-card__actions\\"><button type=\\"button\\" class=\\"v-btn v-btn--text theme--light v-size--default primary--text\\"><span class=\\"v-btn__content\\"> Read more </span></button></div>
      </div>
    </div>
  </div>
</div>"
`;
